packages/i18n-calypso/src/translate-result-to-string.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable jsdoc/no-undefined-types */
+
+/**
+ * Return the string equivalent of a translation result. By translation result, we mean what's
+ * returned by the `translate` function (which is a React element in practice).
+ *
+ * This is helpful when formatted text is used as both tag and attribute content. In the latter
+ * case, HTML must be striped.
+ *
+ * @example
+ * // returns Backup Daily
+ * translateResultToString( translate( 'Backup {{em}}Daily{{/em}}', { components: { em: <em/> } } ) )
+ * @param { TranslateResult } tr Translation returned by the `translate` function
+ * @returns Translation as a string
+ */
+export default function translateResultToString( tr ) {
```

Comment:
Despite the name `translateResultToString`, nothing within the implementation of this helper function suggests that it depends on being used with translation results exclusively.

In fact, the only link to `i18n-calypso` here seems to be the use of the `TranslateResult` type, which could essentially be replaced by [React.ReactChild](https://github.com/Automattic/wp-calypso/blob/9296e86d94b4180f9e12ae780eb398dcb171620d/packages/i18n-calypso/types/index.d.ts#L59).

That being said, I think it should be extracted elsewhere and not be part of the `i18n-calypso` package.

----

On a separate note, I haven't had the chance to test the implementation, but at first glance, it seems like children of type `number` are currently not being handled.

Reply:
Fair enough. I had the same interrogation but given the use cases of this function it felt like home. I'll make its name more generic and move it somewhere else.